include/zephyr/drivers/fuel_gauge.h

```diff
@@ -236,6 +292,35 @@ static inline int z_impl_fuel_gauge_get_prop(const struct device *dev,
 	return api->get_property(dev, props, props_len);
 }
 
+/**
+ * @brief Fetch a battery fuel-gauge block property
+ *
+ * @param dev Pointer to the battery fuel-gauge device
+ * @param prop pointer to single fuel_gauge_get_block_property struct where the property struct
+ * field is set by the caller to determine what property is read from the
+ * fuel gauge device into the dst field.
+ * @param dst byte array or struct that will hold the block data that is read from the fuel gauge
+ * @param dst_len the length of the destination array in bytes
+ *
+ * @return return=0 if successful, return < 0 if getting property failed, return 0 on success
+ */
+__syscall int fuel_gauge_get_block_prop(const struct device *dev,
```

Comment:
I do prefer "buffer" since we're trying to not explicitly tie ourselves to SBS. There are lot of ICs that don't implement SBS but will have similar properties.